src/json_encoding.ml

```diff
@@ -679,21 +679,42 @@ let option : type t. t encoding -> t option encoding = fun t ->
     : type tf. (module Json_repr.Repr with type value = tf) -> tf -> t option
     = fun (module Repr_f) repr ->
       match Repr_f.view repr with
-      | `Null -> None
-      | _ ->
+      | `O [] -> None
```

Comment:
I think I'd rather fail at encoding construction time and keep the previous, simpler scheme.
I think it's find to forbid `option (option _)`.